profiler/lib.profiler/src/org/netbeans/lib/profiler/heap/Systems.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.lib.profiler.heap;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class Systems {
+    private static final Logger LOG = Logger.getLogger(System.class.getPackage().getName());
+    static final boolean DEBUG = LOG.isLoggable(Level.FINE);
```

Comment:
Configurable logging at runtime is one of the key benefits of the logging system to me. Yes there is overhead in the check at runtime, but so is overhead in virtual methods, the VM concept in itself, the interpreter, GC and so on.

I'm not religious here, but reading such code feels strange to me.

Reply:
Let's keep the spirit of the code for now. If needed `DEBUG` can be replaced with `LOG::isLoggable` later.